plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveHadoop.java

```diff
@@ -85,6 +97,25 @@ public HostAndPort getHiveMetastoreEndpoint()
         return getMappedHostAndPortForExposedPort(HIVE_METASTORE_PORT);
     }
 
+    public HiveMetastore getHiveMetastoreClient()
+    {
+        return new BridgingHiveMetastore(new ThriftHiveMetastore(
+                new TestingMetastoreLocator(
+                        Optional.empty(),
+                        getHiveMetastoreEndpoint()),
+                new HiveConfig(),
+                new MetastoreConfig(),
+                new ThriftMetastoreConfig(),
+                new HdfsEnvironment(new HiveHdfsConfiguration(
+                        new HdfsConfigurationInitializer(
+                                new HdfsConfig(),
+                                ImmutableSet.of()),
+                        ImmutableSet.of()),
+                        new HdfsConfig(),
+                        new NoHdfsAuthentication()),
+                false));
```

Comment:
The method creates lots of new objects. For an instance method in HiveHadoop it would be sufficient to expose `getHiveMetastoreEndpoint()`. the rest could be done in the test class itself, or in a reusable utility method.

This would allow the test to choose whether it uses `ThriftHiveMetastore` (should be sufficient) or `BridgingHiveMetastore` wrapper (probably more convenient)

Reply:
Ok moved to test class.